### PR TITLE
add simd headers for sq8uniform

### PR DESCRIPTION
**Pull request by LHT129**

### CODE REVIEW

File: src/simd/sq8_uniform_simd.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "sq8_uniform_simd.h"
+
+#include "simd_status.h"
+
+namespace vsag {
+
+static SQ8UniformComputeCodesType
+SetSQ8UniformComputeCodesIP() {
+    if (SimdStatus::SupportAVX512()) {
+#if defined(ENABLE_AVX512)
```
Comment (jiaweizone) [inline src/simd/sq8_uniform_simd.cpp:25]: Can the values of `SimdStatus::SupportAVX512()` and `ENABLE_AVX512` be different sometimes?

File: src/simd/sq8_uniform_simd.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+
+namespace vsag {
+namespace generic {
+float
+SQ8UniformComputeCodesIP(const uint8_t* codes1, const uint8_t* codes2, uint64_t dim);
```
Comment (jiaweizone) [inline src/simd/sq8_uniform_simd.h:23]: Where is the `SQ8UniformComputeCodesIP` implementation ?

File: src/simd/sq8_uniform_simd.cpp (hunk 1 above)
Comment (LHT129) [reply src/simd/sq8_uniform_simd.cpp:25]: in compile time or at running

File: src/simd/sq8_uniform_simd.h (hunk 2 above)
Comment (LHT129) [reply src/simd/sq8_uniform_simd.h:23]: in another pr, this is headers

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**